File: utils/supabase.ts
import "dotenv/config";
import { createClient } from "@supabase/supabase-js";

const supabaseUrl = process.env.SUPABASE_URL!;
const serviceRoleKey = process.env.SUPABASE_SERVICE_ROLE_KEY!;

if (!supabaseUrl || !serviceRoleKey) {
  throw new Error("SUPABASE_URL or SUPABASE_SERVICE_ROLE_KEY is not set");
}

export const supabase = createClient(supabaseUrl, serviceRoleKey, {
  auth: {
    autoRefreshToken: false,
    persistSession: false,
  },
});

File: utils/supabase.py
import os
from supabase import create_client, Client

def get_supabase():
    """Get Supabase client with proper error handling for missing environment variables."""
    url: str = os.environ.get("SUPABASE_URL")
    key: str = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")
    
    if not url:
        raise ValueError(
            "SUPABASE_URL environment variable is required. "
            "Please set it in your .env file or environment."
        )
    
    if not key:
        raise ValueError(
            "SUPABASE_SERVICE_ROLE_KEY environment variable is required. "
            "Please set it in your .env file or environment."
        )
    
    return create_client(url, key)